trio_asyncio/_loop.py

```diff
@@ -570,7 +614,17 @@ async def forward_cancellation():
                     raise forwarded_cancellation
 
             try:
-                await loop._main_loop_exit()
+                try:
+                    if should_warn:
+                        warnings.warn(
```

Comment:
If the user's warnings filter is configured to raise exceptions instead of printing warnings (`-Werror` mode, fairly common with unit tests) then we want to make sure we still do all the cleanup logic.